plugins/ltac/tactic_debug.ml

```diff
@@ -163,20 +163,20 @@ let cvt_loc loc =
          let file = String.sub dp (lastdot+1) (dplen - (lastdot + 1)) in
          let module_name = String.sub dp 0 lastdot in
          let routine =
-           try
-             (* try text as a kername *)
-             assert (CString.is_prefix dp text);
```

Comment:
Since you stopped using `assert` for flow control do you still need to remove the `when noncritical` guards?

BTW ideally this function should be written without needing any exception catching

Reply:
I can put that back.

It would be nice to have a variant of assert that gives a noncritical exception.

Reply:
no, assert should not be used for flow control

Reply:
Agreed, it should have a distinct name.